pandas/core/arrays/period.py

```diff
@@ -484,9 +484,8 @@ def _time_shift(self, periods, freq=None):
             values[self._isnan] = iNaT
         return type(self)(values, freq=self.freq)
 
-    @property
-    def _box_func(self):
-        return lambda x: Period._from_ordinal(ordinal=x, freq=self.freq)
+    def _box_func(self, x) -> Union[Period, NaTType]:
```

Comment:
should alias this kind of type at somepoint, maybe PeriodOrNaT (and the others as well)